src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -1444,6 +1444,108 @@ public struct S
                 Diagnostic(ErrorCode.ERR_InvalidPropertyReadOnlyMods, "set").WithArguments("S.P8").WithLocation(11, 44));
         }
 
+        [Fact]
+        public void NetModule_ImplicitReadOnlyAutoProperty()
+        {
+            var csharp = @"
+public struct S
+{
+    public int P1 { get; private set; }
+}
+";
+            var moduleMetadata = CreateCompilation(csharp, options: TestOptions.DebugModule, targetFramework: TargetFramework.Mscorlib45).EmitToImageReference();
+            var moduleComp = CreateCompilation("", new[] { moduleMetadata });
+            var moduleGetterAttributes = moduleComp.GetMember<PropertySymbol>("S.P1").GetMethod.GetAttributes();
+            Assert.Equal(1, moduleGetterAttributes.Length);
+            Assert.Equal("CompilerGeneratedAttribute", moduleGetterAttributes[0].AttributeClass.Name);
+
+            var dllMetadata = CreateCompilation(csharp, options: TestOptions.DebugDll, targetFramework: TargetFramework.Mscorlib45).EmitToImageReference();
+            var dllComp = CreateCompilation("", new[] { dllMetadata });
+            var dllGetterAttributes = dllComp.GetMember<PropertySymbol>("S.P1").GetMethod.GetAttributes();
+            Assert.Equal(2, dllGetterAttributes.Length);
+            Assert.Equal("IsReadOnlyAttribute", dllGetterAttributes[0].AttributeClass.Name);
+            Assert.Equal("CompilerGeneratedAttribute", dllGetterAttributes[1].AttributeClass.Name);
+        }
+
+        [Fact]
+        public void NetModule_ImplicitReadOnlyAutoProperty_MalformedAttribute()
+        {
+            var csharp = @"
+namespace System.Runtime.CompilerServices
+{
+    public class IsReadOnlyAttribute
+    {
+        public IsReadOnlyAttribute(int x) {}
+    }
+}
+
+public struct S
+{
+    public int P1 { get; private set; }
+}
+";
+            var moduleComp = CreateCompilation(csharp, options: TestOptions.DebugModule, targetFramework: TargetFramework.Mscorlib45);
+            moduleComp.VerifyDiagnostics(
+                // (12,21): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.IsReadOnlyAttribute..ctor'
+                //     public int P1 { get; private set; }
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "get").WithArguments("System.Runtime.CompilerServices.IsReadOnlyAttribute", ".ctor").WithLocation(12, 21));
```

Comment:
This error doesn't look right -- if no one's actually using the readonly-members feature, they shouldn't get any warnings about the attribute. Maybe you want Compilation.GetWellKnownTypeMember() instead?

Reply:
I think I fixed it. See what you think of the logic I wrote.